applications/visualizer/frontend/src/components/viewers/TwoD/TwoDViewer.tsx

```diff
@@ -0,0 +1,188 @@
+        });
+        cyRef.current = cy;
+
+        const resizeObserver = new ResizeObserver(entries => {
+            for (let entry of entries) {
+                if (entry.target === cyContainer.current) {
+                    updateLayout();
+                }
+            }
+        });
+        resizeObserver.observe(cyContainer.current);
+
+
+        return () => {
+            resizeObserver.disconnect();
+            cy.destroy();
+        };
+    }, []);
+
+    // Fetch and process connections data
+    useEffect(() => {
+        if (!workspace) return;
+
+        // Convert activeNeurons and activeDatasets to comma-separated strings
+        const cells = Array.from(workspace.activeNeurons).join(',');
+        const datasetIds = Object.values(workspace.activeDatasets).map(dataset => dataset.id).join(',');
+        const datasetType = Object.values(workspace.activeDatasets).map(dataset => dataset.type).join(',');
+
+
+        ConnectivityService.getConnections({
+            cells,
+            datasetIds,
+            datasetType,
+            thresholdChemical: thresholdChemical,
+            thresholdElectrical: thresholdElectrical,
+            includeNeighboringCells: includeNeighboringCells,
+            includeAnnotations: includeAnnotations,
+        }).then(connections => {
+            const filteredConnections = filterConnections(connections, workspace, includeNeighboringCells, includeNeighboringCellsAsIndividualCells)
+            setConnections(filteredConnections);
+        }).catch(error => {
+            console.error("Failed to fetch connections:", error);
+        });
+    }, [workspace, includeNeighboringCells, includeNeighboringCellsAsIndividualCells, includeAnnotations, thresholdElectrical, thresholdChemical]);
+
+    // Update graph when connections change
+    useEffect(() => {
+        if (cyRef.current) {
+            updateGraphElements(cyRef.current, connections);
+        }
+    }, [connections]);
+
+    useEffect(() => {
+        if (cyRef.current) {
+            updateNodeColors();
+        }
+    }, [coloringOption]);
+
+    // Update layout when layout setting changes
+    useEffect(() => {
+        updateLayout();
+    }, [layout]);
+
+
+    const updateGraphElements = (cy, connections) => {
+        const nodes = new Set<string>();
+        const edges = [];
+
+        connections.forEach(conn => {
+            nodes.add(conn.pre);
+            nodes.add(conn.post);
+            edges.push(createEdge(conn));
+        });
+
+        const elements = Array.from(nodes).map((nodeId: string) => createNode(nodeId)).concat(edges);
+
+        cy.elements().remove(); // Remove all existing elements
+        cy.add(elements);       // Add new elements
+        updateLayout()
+        updateNodeColors()
+    };
+
+    const updateLayout = () => {
+        if (cyRef.current) {
+            applyLayout(cyRef, layout);
+        }
+    };
+
+    const updateNodeColors = () => {
+        if (cyRef.current) {
```

Comment:
Here also, this is a matter of taste, but writing the negation of the condition with a return shows quickly that the only behavior implemented is when there is a `.current` value (and avoids the nested code)